### PR TITLE
[CORE24-116] task(terraform): copy and adjust terraform files from co…

### PR DESCRIPTION
**Pull request by internetti**
…re v1

Jira ticket: JIRA-CORE24-#

## What have you implemented?
took the terraform files from core v1
these are likely to have to change somehow, but I'll to that in the CD PR

but if anyone already has thoughts here, let them be known

### CODE REVIEW

File: deploy/terraform/azurerm_linux_web_app.app.tf [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  depends_on = [
+    azurerm_role_assignment.acr_pull
+  ]
+}
+
+# AzureRM provider does not yet support the authsettingsv2.
+# For now, we rely on the azapi provider to update the
+# app service with proper oidc settings.
+resource azapi_update_resource app_auth {
```
Comment (neb42) [inline deploy/terraform/azurerm_linux_web_app.app.tf:50]: This block can be replaced by updating the version of terraform. See here: https://github.com/NorwegianRefugeeCouncil/icla-duhok/blob/main/deploy/terraform/azurerm_linux_web_app.app.tf

**Review by neb42 — CHANGES_REQUESTED** [review]

File: deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+resource "azurerm_postgresql_flexible_server" "postgres" {
+  provider               = azurerm.runtime
+  name                   = random_id.postgresql_id.hex
+  resource_group_name    = azurerm_resource_group.rg.name
+  location               = azurerm_resource_group.rg.location
+  version                = "14"
```
Comment (neb42) [inline deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf:6]: variable

File: deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  delegated_subnet_id    = azurerm_subnet.postgres_subnet.id
+  private_dns_zone_id    = azurerm_private_dns_zone.postgres_dns.id
+  administrator_login    = random_pet.postgres_admin_username.id
+  administrator_password = random_password.postgres_admin_password.result
+  zone                   = "1"
```
Comment (neb42) [inline deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf:11]: variable

File: deploy/terraform/azurerm_service_plan.plan.tf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+resource "azurerm_service_plan" "plan" {
+  provider            = azurerm.runtime
+  name                = "asp-${var.app_name}-${var.environment}"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.rg.name
+  os_type             = "Linux"
+  sku_name            = "P1v2"
```
Comment (neb42) [inline deploy/terraform/azurerm_service_plan.plan.tf:7]: variable

File: deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  storage_mb             = 32768
```
[the remark this answers is not in the document]
Comment (internetti) [reply deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf:12]: what about the zone?

File: deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  name                   = random_id.postgresql_id.hex
+  resource_group_name    = azurerm_resource_group.rg.name
+  location               = azurerm_resource_group.rg.location
+  version                = var.postgres_version #"14"
+  delegated_subnet_id    = azurerm_subnet.postgres_subnet.id
+  private_dns_zone_id    = azurerm_private_dns_zone.postgres_dns.id
+  administrator_login    = random_pet.postgres_admin_username.id
+  administrator_password = random_password.postgres_admin_password.result
+  zone                   = var.postgres_zone #"1"
+  storage_mb             = var.postgres_storage_mb #32768
```
Comment (neb42) [inline deploy/terraform/azurerm_postgresql_flexible_server.postgres.tf:12]: Remove the comments

File: deploy/terraform/variables.tf [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+variable "postgres_storage_mb" {
+  type = number
+  description = "MB for postgres storage"
+}
+
+variable "app_service_plan_sku" {
+  type = string
+  description = "SKU for the app service plan"
+}
```
Comment (neb42) [inline deploy/terraform/variables.tf:89]: If we want defaults, they can be added here. I think postgres version can definitely have a default. Tbh they all probably can but we can still configure them if needed.

File: deploy/terraform/azurerm_linux_web_app.app.tf [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  lifecycle {
+    ignore_changes = [
+      site_config.0.application_stack.0.docker_image,
+      site_config.0.application_stack.0.docker_image_tag,
+      app_settings["DOCKER_CUSTOM_IMAGE_NAME"],
+    ]
+  }
+  depends_on = [
+    azurerm_role_assignment.acr_pull
+  ]
```
Comment (neb42) [inline deploy/terraform/azurerm_linux_web_app.app.tf:44]: Not to be super pedantic, but some spaces between each of these blocks would really help readability. I wonder if there's some form of terraform linter?

File: deploy/terraform/main.tf [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.27.0"
+    }
+    azapi = {
+      source  = "Azure/azapi"
+      version = "1.0.0"
+    }
+  }
```
Comment (neb42) [inline deploy/terraform/main.tf:11]: this can go

File: deploy/terraform/main.tf [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  backend "azurerm" {
+  }
+}
+
+provider "azapi" {
+  alias = "runtime"
+}
```
Comment (neb42) [inline deploy/terraform/main.tf:18]: and this

**Review by neb42 — APPROVED** [review]

**Pull request merged**